entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -289,128 +289,20 @@ private void doUpdate(RequestContext requestContext, EntityUpdateRequest request
       AttributeValue attributeValue =
           EntityQueryConverter.convertToAttributeValue(setAttribute.getValue()).build();
       String jsonValue = DocStoreJsonFormat.printer().print(attributeValue);
-      JSONDocument jsonDocument = new JSONDocument(jsonValue);
 
-      Map<Key, Map<String, Document>> entitiesUpdateMap = new HashMap<>();
       for (String entityId : request.getEntityIdsList()) {
         SingleValueKey key =
             new SingleValueKey(requestContext.getTenantId().orElseThrow(), entityId);
-        if (entitiesUpdateMap.containsKey(key)) {
-          entitiesUpdateMap.get(key).put(subDocPath, jsonDocument);
-        } else {
-          Map<String, Document> subDocument = new HashMap<>();
-          subDocument.put(subDocPath, jsonDocument);
-          entitiesUpdateMap.put(key, subDocument);
+        // TODO better error reporting once doc store exposes the,
+        if (!entitiesCollection.updateSubDoc(key, subDocPath, new JSONDocument(jsonValue))) {
+          LOG.warn(
+              "Failed to update entity {}, subDocPath {}, with new doc {}.",
+              key,
+              subDocPath,
+              jsonValue);
         }
       }
-      try {
-        entitiesCollection.bulkUpdateSubDocs(entitiesUpdateMap);
-      } catch (Exception e) {
-        LOG.error(
-            "Failed to update entities {}, subDocPath {}, with new doc {}.",
-            entitiesUpdateMap,
-            subDocPath,
-            jsonValue,
-            e);
-        throw e;
-      }
-    }
-  }
-
-  @Override
-  public void bulkUpdate(
-      BulkEntityUpdateRequest request, StreamObserver<ResultSetChunk> responseObserver) {
-    // Validations
-    RequestContext requestContext = RequestContext.CURRENT.get();
-    Optional<String> maybeTenantId = requestContext.getTenantId();
-    if (maybeTenantId.isEmpty()) {
-      responseObserver.onError(new ServiceException("Tenant id is missing in the request."));
-      return;
-    }
-    if (StringUtils.isEmpty(request.getEntityType())) {
-      responseObserver.onError(new ServiceException("Entity type is missing in the request."));
-      return;
-    }
-    if (request.getEntitiesCount() == 0) {
-      responseObserver.onError(new ServiceException("Entities are missing in the request."));
-    }
-    Map<String, EntityUpdateInfo> entitiesMap = request.getEntitiesMap();
-    try {
-      doBulkUpdate(requestContext, entitiesMap);
-      responseObserver.onCompleted();
```

Comment:
will try to look through this more carefully at some point today, but one thing that jumps out at me off the bat - we never actually `onNext` a result set chunk here that I can see.

Reply:
Discussed a bit offline. There may not be a bug here, but it has a very unintuitive api that can easily lead to corruption - a Value going in looks like a oneof style object but is not, so both the field and type need to be set. If they're not both set, we may be changing data types of persisted data.

Reply:
Oh, the API isn't supposed to send any response, as of now. We can just change the API to just send an empty response as of now, instead of result set chunk

Reply:
If we're going to break the new api by changing its return type anyway, perhaps it makes sense to fix the value type there?